desktop-widgets/locationinformation.cpp

```diff
@@ -134,10 +134,16 @@ void LocationInformationWidget::acceptChanges()
 	struct dive_site *currentDs;
 	uiString = ui.diveSiteName->text().toUtf8().data();
 
-	if (get_dive_site_by_uuid(displayed_dive_site.uuid) != NULL)
+	if (get_dive_site_by_uuid(displayed_dive_site.uuid) != NULL) {
 		currentDs = get_dive_site_by_uuid(displayed_dive_site.uuid);
-	else
-		currentDs = get_dive_site_by_uuid(create_dive_site_from_current_dive(uiString));
+	} else {
+		uint32_t new_uuid = create_dive_site_from_current_dive(uiString);
+		displayed_dive.dive_site_uuid = new_uuid;
+		struct dive *dive_in_table = get_dive_by_uniq_id(displayed_dive.id);
+		if (dive_in_table)
+			dive_in_table->dive_site_uuid = new_uuid;
```

Comment:
So this modifies the dive that is shown, without the normal "accept/discard" policy.
I think that's a bit inconsistent, isn't it?
I have been working on this but for a while and my fix isn't quite ready yet - or I should say I have fixed part of the problem but not all of it. I'll create a PR from what I have done so far for you to take a look at :-)

Reply:
seems to confirm my `i don't think this is the right fix; please have a look...`.